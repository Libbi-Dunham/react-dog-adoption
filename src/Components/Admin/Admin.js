import React, { useState } from 'react';
import { addDog } from '../../services/DogRoute';
import DogForm from '../../Components/DogCard/DogForm';
import { useHistory } from 'react-router-dom';

export default function CreateDog() {
  const [name, setName] = useState('');
  const [bio, setBio] = useState('');
  const [image, setImage] = useState('');
  const [age, setAge] = useState('');
  const [breed, setBreed] = useState('');
  const history = useHistory();

  const handleSubmit = async (e) => {
    try {
      alert('You Have Successfully Added A dog!');
    } catch {
      alert('Oh No You Have No Success!');
    }
    e.preventDefault();
    await addDog(name, bio, image, age, breed);
    history.push('/');
  };

  return (
    <div>
      <h1>Add a new Dog</h1>
      <DogForm
        name={name}
        setName={setName}
        bio={bio}
        setBio={setBio}
        image={image}
        setImage={setImage}
        age={age}
        setAge={setAge}
        breed={breed}
        setBreed={setBreed}
        handleSubmit={handleSubmit}
      />
    </div>
  );
}
