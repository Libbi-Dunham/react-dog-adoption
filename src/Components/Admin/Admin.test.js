import { render } from '@testing-library/react';
import Admin from './Admin';

test('renders the admin page/new dog', () => {
  const container = render(<Admin />);
  expect(container).toMatchSnapshot();
});
