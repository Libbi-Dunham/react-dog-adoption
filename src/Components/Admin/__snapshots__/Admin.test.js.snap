// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the admin page/new dog 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          Add a new Dog
        </h1>
        <div
          class="dog-form"
        >
          <form>
            <h3>
              Enter A New Dog
            </h3>
            <div
              class="form-control"
            >
              <label>
                Name: 
              </label>
              <input
                class="control"
                placeholder="Name"
                type="text"
                value=""
              />
            </div>
            <div
              classsname="form-control"
            >
              <label>
                Bio: 
              </label>
              <textarea
                class="control"
                placeholder="bio"
                type="text"
              />
            </div>
            <div
              class="form-control"
            >
              <label>
                Age: 
              </label>
              <input
                class="control"
                placeholder="number"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-control"
            >
              <label>
                Image: 
              </label>
              <input
                class="control"
                placeholder="Image URL"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-control"
            >
              <label>
                Breed: 
              </label>
              <input
                class="control"
                placeholder="breed"
                type="text"
                value=""
              />
            </div>
            <br />
            <br />
            <div
              class="action"
            >
              <button
                class="save"
              >
                Save
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        Add a new Dog
      </h1>
      <div
        class="dog-form"
      >
        <form>
          <h3>
            Enter A New Dog
          </h3>
          <div
            class="form-control"
          >
            <label>
              Name: 
            </label>
            <input
              class="control"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
          <div
            classsname="form-control"
          >
            <label>
              Bio: 
            </label>
            <textarea
              class="control"
              placeholder="bio"
              type="text"
            />
          </div>
          <div
            class="form-control"
          >
            <label>
              Age: 
            </label>
            <input
              class="control"
              placeholder="number"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-control"
          >
            <label>
              Image: 
            </label>
            <input
              class="control"
              placeholder="Image URL"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-control"
          >
            <label>
              Breed: 
            </label>
            <input
              class="control"
              placeholder="breed"
              type="text"
              value=""
            />
          </div>
          <br />
          <br />
          <div
            class="action"
          >
            <button
              class="save"
            >
              Save
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
